### PR TITLE
CRM-16408 WordPress Installer changes 

### PR DESCRIPTION
**Pull request by kcristiano**
REplaces #6629

---
- [CRM-16408: Installer rewrite](https://issues.civicrm.org/jira/browse/CRM-16408)

### CODE REVIEW

File: install/index.php [hunk 1]
```diff
@@ -501,8 +510,13 @@ public function check() {
       );
     }
     elseif ($installType == 'wordpress') {
-      // make sure that we can write to plugins/civicrm  and plugins/files/
-      $writableDirectories = array(WP_PLUGIN_DIR . DIRECTORY_SEPARATOR . 'files', $cmsPath);
+      // make sure that we can write to uploads/civicrm/
+      $upload_dir = wp_upload_dir();
+      $files_dirname = $upload_dir[basedir] . DIRECTORY_SEPARATOR . 'civicrm';
```
Comment (totten) [inline install/index.php]: Quotes

File: install/index.php [hunk 2]
```diff
@@ -107,12 +107,19 @@
...
-    "password" => "",
-    "database" => "civicrm",
-  );
+  if ($installType == 'wordpress') {
+    //WP Database Data
+    $database = DB_NAME;
+    $username = DB_USER;
+    $password = DB_PASSWORD;
+    $server = DB_HOST;
+    $databaseConfig = array(
```
Comment (totten) [inline install/index.php]: Tighten (inline $database)

File: CRM/Utils/File.php [hunk 3]
```diff
@@ -465,6 +465,34 @@ public static function restrictAccess($dir, $overwrite = FALSE) {
...
+   *   The directory to be secured.
+   * @param bool $overwrite
+   */
+  public static function restrictAccessSettings($dir, $overwrite = FALSE) {
+    // note: empty value for $dir can play havoc, since that might result in putting '.htaccess' to root dir
+    // of site, causing site to stop functioning.
+    // FIXME: we should do more checks here -
+    if (!empty($dir) && is_dir($dir)) {
+      $htaccess = <<<HTACCESS
+<Files "civicrm.settings.php">
```
Comment (totten) [inline CRM/Utils/File.php]: This seemed a bit extravagant.

**Comment by totten** [comment]
We also discussed updating `getDefaultFileStorage()` in `CRM_Utils_System_Base` / `CRM_Utils_System_WordPress`.

**Comment by kcristiano** [comment]
Dependency  :   https://github.com/civicrm/civicrm-wordpress/pull/85

**Comment by colemanw** [comment]
@kcristiano do you feel like squashing some of these commits together? Looks like the first 3 could be squashed, as well as the 2 "fix syntax" ones.

**Comment by kcristiano** [comment]
@colemanw I can give it a try.   This was more for me to keep everything straight after a bad rebase (my error)

**Comment by kcristiano** [comment]
@colemanw I gave it a try, not sure if I accomplished what you wanted.  Let me know.

**Comment by colemanw** [comment]
err, it appears that instead of fewer commits there are now more.
My suggestion was along the lines of doing this:

    git checkout 16408-a
    git rebase -i 5c09857

You'd then be prompted (via a text file) to say what you want to do with each commit. I'd say squash most of them.

If all goes well, after the rebase you can type `git push -f` which will overwrite the contents of this PR with your new revised commit history. That's what I do, but since this is your first time playing with this you might instead want to create a new branch and open a new PR instead, which would be a bit safer.

**Comment by kcristiano** [comment]
@colemanw Yes, this has gone bad.  And as there are other commits that have been merged to these files I will need to create a new branch and start again.

Apologies for my lack of git expertise, let me see when I can get to this.

**Comment by kcristiano** [comment]
superseded by https://github.com/civicrm/civicrm-core/pull/6907

**Pull request closed without merging**